broker-daemon/broker-rpc/wallet-service/commit.js

```diff
@@ -47,53 +45,82 @@ async function commit ({ params, relayer, logger, engines, orderbooks }, { Empty
-  } else if (maxChannelBalance.lt(balance)) {
-    const maxChannelBalanceCommon = Big(maxChannelBalance).div(engine.quantumsPerCommon).toString()
-    logger.error(`Balance from the client exceeds maximum balance allowed (${maxChannelBalance.toString()}).`, { balance })
-    throw new Error(`Maximum balance of ${maxChannelBalanceCommon} ${symbol} exceeded for ` +
-      `committing of ${balanceCommon} ${symbol} to the Relayer. Please try again.`)
   }
 
-  // Get the max balance for outbound channel to see if there are already channels with the balance open. If this is the
-  // case we do not need to go to the trouble of opening a new channel
-  const { maxBalance: maxOutboundBalance } = await engine.getMaxChannel()
+  // Collect all remote information
+  const [
+    { address: relayerAddress },
+    { address: relayerInverseAddress },
+    outboundPaymentChannelNetworkAddress,
+    inboundPaymentChannelNetworkAddress,
+    unspentBalance
+  ] = await Promise.all([
+    relayer.paymentChannelNetworkService.getAddress({ symbol }),
+    relayer.paymentChannelNetworkService.getAddress({ symbol: inverseSymbol }),
+    engine.getPaymentChannelNetworkAddress(),
+    inverseEngine.getPaymentChannelNetworkAddress(),
+    engine.getConfirmedBalance()
+  ])
+
+  logger.info(`Attempting to create channel with ${relayerAddress} on ${symbol} with ${balanceCommon}`, {
+    balanceCommon,
+    balance
+  })
 
-  // If maxOutboundBalance exists, we need to check if the balance is greater or less than the balance of the channel
-  // we are trying to open. If maxOutboundBalance does not exist, it means there are no channels open and we can safely
-  // attempt to create channels with the balance
-  if (maxOutboundBalance) {
-    const insufficientOutboundBalance = maxOutboundBalance && Big(maxOutboundBalance).plus(engine.feeEstimate).lt(balance)
+  const currentBalance = await engine.getTotalBalanceForAddress(relayerAddress)
+  let balanceToCommit = Big(balance).minus(currentBalance)
 
-    if (insufficientOutboundBalance) {
-      logger.error('Existing outbound channel of insufficient size', { desiredBalance: balance, feeEstimate: engine.feeEstimate, existingBalance: maxOutboundBalance })
-      throw new Error('You have an existing outbound channel with a balance lower than desired, release that channel and try again.')
+  if (balanceToCommit.lte(0)) {
+    logger.debug('Channels with sufficient balance already exist', { balance: currentBalance })
+  } else {
+    // Increase the balance of the final channel if it would be uneconomic
+    const lastChannelBalance = balanceToCommit.mod(engine.maxChannelBalance)
+    if (lastChannelBalance.gt(0) && lastChannelBalance.lt(MINIMUM_FUNDING_AMOUNT.times(engine.quantumsPerCommon))) {
+      logger.error('Minimum channel balance would have resulted in an uneconomic channel.', { balanceToCommit, lastChannelBalance })
+
+      if (lastChannelBalance.lte(engine.feeEstimate)) {
```

Comment:
would add a comment above this just because it is harder to parse the error messages on first glance